packages/file_selector/file_selector_ios/lib/file_selector_ios.dart

```diff
@@ -0,0 +1,120 @@
+
+  /// Registers the iOS implementation.
+  static void registerWith() {
+    FileSelectorPlatform.instance = FileSelectorIOS();
+  }
+
+  @override
+  Future<XFile?> openFile({
+    List<XTypeGroup>? acceptedTypeGroups,
+    String? initialDirectory,
+    String? confirmButtonText,
+  }) async {
+    final List<String>? path = await _channel.invokeListMethod<String>(
+      'openFile',
+      <String, dynamic>{
+        'acceptedTypes': _allowedTypeListFromTypeGroups(acceptedTypeGroups),
+        'initialDirectory': initialDirectory,
+        'confirmButtonText': confirmButtonText,
+        'multiple': false,
+      },
+    );
+    return path == null ? null : XFile(path.first);
+  }
+
+  @override
+  Future<List<XFile>> openFiles({
+    List<XTypeGroup>? acceptedTypeGroups,
+    String? initialDirectory,
+    String? confirmButtonText,
+  }) async {
+    final List<String>? pathList = await _channel.invokeListMethod<String>(
+      'openFile',
+      <String, dynamic>{
+        'acceptedTypes': _allowedTypeListFromTypeGroups(acceptedTypeGroups),
+        'initialDirectory': initialDirectory,
+        'confirmButtonText': confirmButtonText,
+        'multiple': true,
+      },
+    );
+    return pathList?.map((String path) => XFile(path)).toList() ?? <XFile>[];
+  }
+
+  @override
+  Future<String?> getSavePath({
+    List<XTypeGroup>? acceptedTypeGroups,
+    String? initialDirectory,
+    String? suggestedName,
+    String? confirmButtonText,
+  }) async {
+    return _channel.invokeMethod<String>(
+      'getSavePath',
+      <String, dynamic>{
+        'acceptedTypes': _allowedTypeListFromTypeGroups(acceptedTypeGroups),
+        'initialDirectory': initialDirectory,
+        'suggestedName': suggestedName,
+        'confirmButtonText': confirmButtonText,
+      },
+    );
+  }
+
+  @override
+  Future<String?> getDirectoryPath({
+    String? initialDirectory,
+    String? confirmButtonText,
+  }) async {
+    return _channel.invokeMethod<String>(
+      'getDirectoryPath',
+      <String, dynamic>{
+        'initialDirectory': initialDirectory,
+        'confirmButtonText': confirmButtonText,
+      },
+    );
+  }
+
+  // Converts the type group list into a flat list of all allowed types, since
+  // iOS doesn't support filter groups.
+  Map<String, List<String>>? _allowedTypeListFromTypeGroups(
+      List<XTypeGroup>? typeGroups) {
+    const String extensionKey = 'extensions';
+    const String mimeTypeKey = 'mimeTypes';
+    const String utiKey = 'UTIs';
+    if (typeGroups == null || typeGroups.isEmpty) {
+      return null;
+    }
+    final Map<String, List<String>> allowedTypes = <String, List<String>>{
+      extensionKey: <String>[],
+      mimeTypeKey: <String>[],
+      utiKey: <String>[],
+    };
+    for (final XTypeGroup typeGroup in typeGroups) {
+      // If any group allows everything, no filtering should be done.
+      if ((typeGroup.extensions?.isEmpty ?? true) &&
+          (typeGroup.macUTIs?.isEmpty ?? true) &&
```

Comment:
This turns out to have been poor naming :( Filed https://github.com/flutter/flutter/issues/103743

(Doesn't need to block this PR, just leaving here for future reference.)